From c40c18d9d69445223d110575a33b7c0f38022ded Mon Sep 17 00:00:00 2001
From: John Williams <john.williams@petalogix.com>
Date: Thu, 10 Nov 2011 14:39:15 +1000
Subject: [PATCH] serial: uartlite: Warn on unconnected IRQ

Give some warning if a uartlite IP is missing the IRQ before aborting the probe.

Signed-off-by: John Williams <john.williams@petalogix.com>
Signed-off-by: Michal Simek <monstr@monstr.eu>
---
 drivers/serial/uartlite.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/drivers/serial/uartlite.c b/drivers/serial/uartlite.c
index 76b5f81..6cb1e9d 100644
--- a/drivers/serial/uartlite.c
+++ b/drivers/serial/uartlite.c
@@ -580,8 +580,10 @@ static int __devinit ulite_probe(struct platform_device *pdev)
 		return -ENODEV;
 
 	res2 = platform_get_resource(pdev, IORESOURCE_IRQ, 0);
-	if (!res2)
+	if (!res2) {
+		dev_warn(&pdev->dev, "refusing to probe uart instance with no IRQ");
 		return -ENODEV;
+	}
 
 	return ulite_assign(&pdev->dev, pdev->id, res->start, res2->start);
 }
@@ -625,6 +627,7 @@ ulite_of_probe(struct platform_device *op, const struct of_device_id *match)
 
 	irq = irq_of_parse_and_map(op->dev.of_node, 0);
 	if (irq == -1) {
+		dev_warn(&op->dev, "refusing to probe uart instance with no IRQ");
 		return -ENODEV;
 	}
 
-- 
1.7.0.4

